data/data-pipeline/data_pipeline/comparison_tool/src/tract_comparison__template.ipynb

```diff
@@ -0,0 +1,379 @@
+   "source": [
+    "## These are parameters and get overridden by the \"injected parameters\" cell below\n",
+    "ADDITIONAL_DEMO_COLUMNS = []\n",
+    "COMPARATOR_COLUMN = None\n",
+    "COMPARATOR_FILE = None\n",
+    "DEMOGRAPHIC_COLUMNS = []\n",
+    "DEMOGRAPHIC_FILE = None\n",
+    "OUTPUT_DATA_PATH = None\n",
+    "SCORE_FILE = None\n",
+    "OTHER_COMPARATOR_COLUMNS = None\n",
+    "OUTPUT_NAME = None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bd38aaaa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## These are constants for all runs\n",
+    "GEOID_COLUMN = field_names.GEOID_TRACT_FIELD\n",
+    "SCORE_COLUMN = field_names.SCORE_M_COMMUNITIES\n",
+    "TOTAL_POPULATION_COLUMN = field_names.TOTAL_POP_FIELD"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a3940be",
+   "metadata": {},
+   "source": [
+    "__Date and time of last run__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5f19a1e3-fe05-425d-9160-4d4e193fadf7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datetime.datetime.now()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "269b0f06-edee-4fce-8755-a8e7fe92e340",
+   "metadata": {},
+   "source": [
+    "__Congfigure output (autocreated)__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "acace944-18fa-4b05-a581-8147e8e09299",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "OUTPUT_EXCEL = os.path.join(\n",
+    "    OUTPUT_DATA_PATH,\n",
+    "    f\"{OUTPUT_NAME}__{datetime.datetime.now().strftime('%Y-%m-%d')}.xlsx\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f5517217-f107-4fee-b4ca-be61f6b2b7c3",
+   "metadata": {},
+   "source": [
+    "__Read in data__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cdefecf2-100b-4ad2-b0bd-101c0bad5a92",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "comparator_cols = [COMPARATOR_COLUMN] + OTHER_COMPARATOR_COLUMNS if OTHER_COMPARATOR_COLUMNS else [COMPARATOR_COLUMN]\n",
+    "\n",
+    "#papermill_description=Loading_data\n",
+    "joined_frame = pd.concat(\n",
+    "    [\n",
+    "        utils.read_file(\n",
+    "            SCORE_FILE,\n",
+    "            columns=[TOTAL_POPULATION_COLUMN, SCORE_COLUMN] + ADDITIONAL_DEMO_COLUMNS,\n",
+    "            geoid=GEOID_COLUMN,\n",
+    "        ),\n",
+    "        utils.read_file(COMPARATOR_FILE, columns=comparator_cols, geoid=GEOID_COLUMN),\n",
+    "        utils.read_file(\n",
+    "            DEMOGRAPHIC_FILE,\n",
+    "            DEMOGRAPHIC_COLUMNS,\n",
+    "            geoid=GEOID_COLUMN,\n",
+    "        ),\n",
+    "    ],\n",
+    "    axis=1,\n",
+    ").reset_index()\n",
+    "\n",
+    "# Comparator may provide simple list of tracts; this will fill all census tracts with \"False\"\n",
+    "joined_frame = joined_frame.fillna({COMPARATOR_COLUMN: False, SCORE_COLUMN: False})"
```

Comment:
This is interesting! A few thoughts here: 

1. I think the case I'm thinking through here is the "lists of DACS" -- so like, for the rural partners analysis, we only got which tracts WERE classified as DACS, not a list of all tracts with a boolean TRUE / FALSE. I am wondering whether `np.nan` is appropriate in that case for tracts that aren't identified? In that case, we'd fill `False` any tract for a state that is included, and force that tract into "it's been considered by the tool". Just wanted to note that caveat -- then for the "segmented analysis", we'd drop all nulls (as in `df.dropna(subset=[SCORE_COLUMN, COMPARATOR_COLUMN], how="any")` with the idea that we'd only be comparing the disjoint set among *eligible* tracts? (I'd still vote to keep the "overall" statistics for CEJST with nulls -- so then the overall would have "Comparator, not comparator, not considered" for the comparator) 

2. For this -- `For instance, this would be helpful when comparing a score with fixes for missing income data to Definition M.` -- I think these are ALREADY cast as booleans for Score M fields -- like, we only allow something to be or not to be a DAC, not to have a null value for being a DAC.

Reply:
(i updated this i think!)